pkg/nodeconfig/nodeconfig.go

```diff
@@ -265,6 +276,63 @@ func (nc *nodeConfig) applyLabelsAndAnnotations(labels, annotations map[string]s
 	return nil
 }
 
+// configureAzureCloudNodeManager configures azure cloud node manager
+func (nc *nodeConfig) configureAzureCloudNodeManager() error {
+	cfg, err := crclientcfg.GetConfig()
+	if err != nil {
+		return errors.Wrap(err, "unable to get config to talk to kubernetes api server")
+	}
+
+	client, err := clientset.NewForConfig(cfg)
+	if err != nil {
+		return errors.Wrap(err, "unable to get client from the given config")
+	}
+
+	infra, err := client.ConfigV1().Infrastructures().Get(context.TODO(), "cluster", meta.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, "unable to get cluster infrastructure resource")
+	}
+
+	// We don't need to deploy Azure Cloud Node Manager if platform is not Azure
+	if infra.Status.PlatformStatus.Type != configv1.AzurePlatformType {
+		return nil
+	}
+
+	// We don't need to deploy the service if Cloud Controller is not owned by Cloud Controller Manager
+	ownedByCCM, err := isCloudControllersOwnedByCCM(client)
+	if err != nil {
+		return err
+	} else if !ownedByCCM {
+		return nil
+	}
```

Comment:
This is going to be run on every node configuration. Looks like this decision can be a one time operation on operator boot. We can cache the result i.e. ownedByCCM and use it here. This could be done in the [cluster](https://github.com/openshift/windows-machine-config-operator/blob/master/pkg/cluster/config.go#L1) package. WDYT?

Reply:
Yes, it should be run on every node configuration, because CloudController ownership can be changed during the cluster lifecycle. For example, currently by default we deploy clusters where KCM is the owner, and therefore isCloudControllersOwnedByCCM returns False. Then users may explicitly set ownership to CCM by modifying the cluster FeatureGate, and the function will return True. Anyway, this is a temporary solution, soon Cloud Controllers will always be owned by CCM.

Reply:
OK, then it makes sense to call it on each node configuration. One suggestion I have is to move the the code that reaches out to the cluster to figure out the configuration to the cluster package. IOW, something to the effect of:
```
if cluster.CloudControllerOwnedByCCM() {
    if err := nc.Windows.ConfigureAzureCloudNodeManager(nc.node.GetName()); err != nil {
		return err
	}
}
```
That way we clearly separate out the cluster specific configuration checking into the cluster package. WDYT?

Reply:
yeah, it makes sense. done :)